test/pkg/environment/common/expectations.go

```diff
@@ -0,0 +1,927 @@
+			return e.Name == v.Name
+		}); ok {
+			d.Spec.Template.Spec.Containers[0].Env[i] = v
+		} else {
+			d.Spec.Template.Spec.Containers[0].Env = append(d.Spec.Template.Spec.Containers[0].Env, v)
+		}
+	}
+	if !equality.Semantic.DeepEqual(d, stored) {
+		By("overriding environment variables for karpenter deployment")
+		Expect(env.Client.Patch(env.Context, d, client.StrategicMergeFrom(stored))).To(Succeed())
+		env.EventuallyExpectKarpenterRestarted()
+	}
+}
+
+func (env *Environment) ExpectSettingsRemoved(vars ...v1.EnvVar) {
+	GinkgoHelper()
+
+	varNames := sets.New[string](lo.Map(vars, func(v v1.EnvVar, _ int) string { return v.Name })...)
+
+	d := &appsv1.Deployment{}
+	Expect(env.Client.Get(env.Context, types.NamespacedName{Namespace: "kube-system", Name: "karpenter"}, d)).To(Succeed())
+	Expect(d.Spec.Template.Spec.Containers).To(HaveLen(1))
+
+	stored := d.DeepCopy()
+	d.Spec.Template.Spec.Containers[0].Env = lo.Reject(d.Spec.Template.Spec.Containers[0].Env, func(v v1.EnvVar, _ int) bool {
+		return varNames.Has(v.Name)
+	})
+	if !equality.Semantic.DeepEqual(d, stored) {
+		By("removing environment variables for karpenter deployment")
+		Expect(env.Client.Patch(env.Context, d, client.StrategicMergeFrom(stored))).To(Succeed())
+		env.EventuallyExpectKarpenterRestarted()
+	}
+}
+
+func (env *Environment) ExpectConfigMapExists(key types.NamespacedName) *v1.ConfigMap {
+	GinkgoHelper()
+	cm := &v1.ConfigMap{}
+	Expect(env.Client.Get(env, key, cm)).To(Succeed())
+	return cm
+}
+
+func (env *Environment) ExpectConfigMapDataReplaced(key types.NamespacedName, data ...map[string]string) (changed bool) {
+	GinkgoHelper()
+	cm := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      key.Name,
+			Namespace: key.Namespace,
+		},
+	}
+	err := env.Client.Get(env, key, cm)
+	Expect(client.IgnoreNotFound(err)).ToNot(HaveOccurred())
+
+	stored := cm.DeepCopy()
+	cm.Data = lo.Assign(data...) // Completely replace the data
+
+	// If the data hasn't changed, we can just return and not update anything
+	if equality.Semantic.DeepEqual(stored, cm) {
+		return false
+	}
+	// Update the configMap to update the settings
+	env.ExpectCreatedOrUpdated(cm)
+	return true
+}
+
+func (env *Environment) ExpectConfigMapDataOverridden(key types.NamespacedName, data ...map[string]string) (changed bool) {
+	GinkgoHelper()
+	cm := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      key.Name,
+			Namespace: key.Namespace,
+		},
+	}
+	err := env.Client.Get(env, key, cm)
+	Expect(client.IgnoreNotFound(err)).ToNot(HaveOccurred())
+
+	stored := cm.DeepCopy()
+	cm.Data = lo.Assign(append([]map[string]string{cm.Data}, data...)...)
+
+	// If the data hasn't changed, we can just return and not update anything
+	if equality.Semantic.DeepEqual(stored, cm) {
+		return false
+	}
+	// Update the configMap to update the settings
+	env.ExpectCreatedOrUpdated(cm)
+	return true
+}
+
+func (env *Environment) ExpectPodENIEnabled() {
```

Comment:
Looking at a few of these functions, I'm realizing that we may have leaked some of the AWS expectations into this common package. Can we pull them out and then update our E2E tests in `aws/karpenter-provider-aws` so that we are importing these common expectations and not duplicating the code over in provider?